src/main/java/com/iota/iri/hash/Kerl.java

```diff
@@ -11,8 +11,15 @@
     public static final int BIT_HASH_LENGTH = 384;
     public static final int BYTE_HASH_LENGTH = BIT_HASH_LENGTH / 8;
 
-    private byte[] byte_state;
-    private int[] trit_state;
+    // radix constant conversion for fast math
+    public static final BigInteger RADIX = BigInteger.valueOf(Converter.RADIX);
+    // lookups for fast math
+    private static final BigInteger[] RADIX_POWERS = makeRadixPowers();
+    private static final BigInteger[] RADIX_POSITIVE_ONE_FUNCTION = makeRadixFunctionPowers(true);
+    private static final BigInteger[] RADIX_NEGATIVE_ONE_FUNCTION = makeRadixFunctionPowers(false);
+
```

Comment:
The radix positive one function is a prepopulated lookup so that repeat trits of 1 or -1 are easily converted without multiple round-trips of `value = value.multiply(RADIX).add(ONE)`

In the first index is 1.
The second is B + 1
The third is B^2 + B + 1
The fourth is B^3 + B^2 + B + 1
etc.

The negative one is:

- AB^2 - B - 1
- AB^3 - B^2 - B - 1 
etc.

There are many instances of repeat number in the trits array and this saves both the round tripping and a lot of instances of BigInteger.